# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1382028617 14400
#      Thu Oct 17 12:50:17 2013 -0400
# Node ID dd125b5e809064cf0350db4aced79e847410abdb
# Parent  9f6e4e5c2bac5b7933be27415086bb732ffb1fdb
handle OCTAVE_HOME substitution for NEWS file (bug #40284)

* main-window.cc (main_window::display_release_notes): Use
Voct_etc_dir, which accounts for OCTAVE_HOME, instead of
OCTAVE_OCTETCDIR.

diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -246,17 +246,19 @@ void
 main_window::open_online_documentation_page (void)
 {
   QDesktopServices::openUrl (QUrl ("http://octave.org/doc/interpreter"));
 }
 
 void
 main_window::display_release_notes (void)
 {
-  display_url_in_window (QUrl ("file://" OCTAVE_OCTETCDIR "/NEWS"));
+  std::string news_file = "file://" + Voct_etc_dir + "/NEWS";
+
+  display_url_in_window (QUrl (QString::fromStdString (news_file)));
 }
 
 void
 main_window::display_url_in_window (const QUrl& url)
 {
   QWidget *w = new QWidget;
 
   QTextBrowser *browser = new QTextBrowser (w);
